adders/tailwindcss/config/adder.ts

```diff
@@ -71,19 +71,30 @@ export const adder = defineAdderConfig({
         {
             name: () => "src/app.css",
             contentType: "css",
-            content: ({ ast, addAtRule }) => {
-                // `addAtRule` prepends the rules to the tree, so we reverse the order
-                // such that the last element ends up at the top of the stylesheet
-                const imports = ['"tailwindcss/utilities"', '"tailwindcss/components"', '"tailwindcss/base"'];
+            content: ({ ast, AtRule }) => {
+                const layers = ["base", "components", "utilities"];
                 const originalFirst = ast.first;
-                const nodes = imports.map((name) => addAtRule(ast, "import", name));
+
+                let prev: typeof originalFirst;
+                const nodes = layers.map((layer) => {
+                    const param = `"tailwindcss/${layer}"`;
+                    const found = ast.nodes.find((x) => x.type === "atrule" && x.name === "import" && x.params === param);
+
+                    if (found) return (prev = found);
+
+                    const rule = new AtRule({ name: "import", params: param });
+                    if (prev) ast.insertAfter(prev, rule);
+                    else ast.prepend(rule);
+
+                    return (prev = rule);
+                });
 
                 if (originalFirst !== ast.first && originalFirst?.type === "atrule" && originalFirst.name === "import") {
                     originalFirst.raws.before = "\n";
                 }
 
-                // We remove the last node to avoid adding a newline at the top of the stylesheet
-                nodes.pop();
+                // We remove the first node to avoid adding a newline at the top of the stylesheet
+                nodes.shift();
 
                 // Each node is prefixed with single newline, ensuring the imports will always be single spaced.
```

Comment:
it looks like we're not using `addAtRule` anymore, but I wonder if this logic would make sense to have live in there or some common util function?

Reply:
we certainly could! probably in the form of an `addImport` method

Reply:
Could the `before = "\n"` part live there too?

Reply:
it _could_, but i'd rather have the caller have agency over those specifics for now - perhaps if another adder requires it in the future, it could be added there as well

Reply:
Double spacing seems like something you'd rarely want to do though. I imagine if you set the default to single spacing that the caller could then override as needed. There could potentially be some option to control spacing as another option - though I'm not really sure what you'd want besides single-spaced by default and then let the caller override as needed